File: solutions/fsharp/raindrops/1/Raindrops.fs
module Raindrops

let dropName divisor divisorName (number, fullName) =
    if number % divisor = 0 then
        (number, fullName + divisorName)
    else
        (number, fullName)

let unwrapDrop (number, fullName) =
    if String.length fullName > 0 then
        fullName
    else
        number.ToString()

let convert (number: int): string =
    (number, "")
    |> dropName 3 "Pling"
    |> dropName 5 "Plang"
    |> dropName 7 "Plong"
    |> unwrapDrop

File: solutions/csharp/raindrops/1/Raindrops.cs
using System.Collections.Generic;
using System.Linq;

public static class Raindrops
{
    private static IEnumerable<(int, string)> _drops = new List<(int, string)>{
      (3, "Pling"),
      (5, "Plang"),
      (7, "Plong")
    };

    public static string Convert(int number)
    {
        var processed = _drops.Aggregate("", (l, i) => number % i.Item1 == 0 ? l + i.Item2 : l);
        return processed.Length > 0 ? processed : number.ToString();
    }
}